python/cudf/cudf/core/dataframe.py

```diff
@@ -2566,34 +2542,14 @@ def groupby(
             raise TypeError(
                 "groupby() requires either by or level to be" "specified."
             )
-        if method == "cudf":
-            from cudf.core.groupby.legacy_groupby import Groupby
-
-            if as_index:
-                warnings.warn(
-                    "as_index==True not supported due to the lack of "
-                    "multi-index with legacy groupby function. Use hash "
-                    "method for multi-index"
-                )
-            result = Groupby(self, by=by)
-            return result
-        else:
-            from cudf.core.groupby.groupby import DataFrameGroupBy
 
-            # The corresponding pop() is in
-            # DataFrameGroupBy._apply_aggregation()
-            libcudf.nvtx.nvtx_range_push("CUDF_GROUPBY", "purple")
-
-            result = DataFrameGroupBy(
-                self,
-                by=by,
-                method=method,
-                as_index=as_index,
-                sort=sort,
-                level=level,
-                dropna=dropna,
+        if method is not None:
+            warnings.warn(
+                "The 'method' argument is deprecated and will be unused"
```

Comment:
Define this as a `DeprecationWarning`: https://docs.python.org/3.6/library/exceptions.html#DeprecationWarning